src/shared/components/common/markdown-textarea.tsx

```diff
@@ -346,60 +348,74 @@ export class MarkdownTextArea extends Component<
   }
 
   handleImageUpload(i: MarkdownTextArea, event: any) {
-    let file: any;
-    if (event.target) {
-      event.preventDefault();
-      file = event.target.files[0];
-    } else {
-      file = event;
+    i.setState({ imageLoading: true });
+    // TODO: not sure if this is needed
+    if (!event.target) {
+      console.log("event");
+      i.uploadImage(i, event)
+      i.setState({ imageLoading: false });
+      return
+    } 
+    if (event.target.files.length > 20) {
+        toast(i18n.t("too_many_images_upload"), "danger");
+        i.setState({ imageLoading: false });
+        return
     }
-
+    const processFiles = async () => {
```

Comment:
Don't use var lambdas here, make this an async function.

Reply:
alright, done.